addon/ng2/blueprints/component/files/__path__/__name__.ts

```diff
@@ -2,12 +2,10 @@ import {Component} from 'angular2/core';
 
 
 @Component({
+  moduleId: __moduleName,
   selector: '<%= dasherizedModuleName %>',
-  templateUrl: 'app<%= dynamicPath %>/<%= dasherizedModuleName %>/<%= dasherizedModuleName %>.html',
-  styleUrls: ['app<%= dynamicPath %>/<%= dasherizedModuleName %>/<%= dasherizedModuleName %>.css'],
-  providers: [],
-  directives: [],
-  pipes: []
+  templateUrl: '<%= dasherizedModuleName %>.html',
+  styleUrls: ['<%= dasherizedModuleName %>.css']
```

Comment:
Angular itself uses the `moduleId` to build the path for templates and styles. SystemJS adds the `__moduleName` to the compiled files. It works.